src/vs/workbench/contrib/terminalContrib/suggest/browser/lspCompletionProviderAddon.ts

```diff
@@ -23,24 +24,45 @@ export class LspCompletionProviderAddon extends Disposable implements ITerminalA
 	readonly triggerCharacters?: string[] | undefined;
 	private _provider: any;
 	private _textVirtualModel: IReference<IResolvedTextEditorModel>;
-
+	private _commandDetection: ICommandDetectionCapability | undefined;
+	private _capabilitiesStore: ITerminalCapabilityStore;
 	constructor(
 		provider: CompletionItemProvider,
 		textVirtualModel: IReference<IResolvedTextEditorModel>,
+		capabilityStore: ITerminalCapabilityStore,
 		// triggerCharacters: string[] | undefined,
 		// id: string,
 	) {
 		super();
+		this._capabilitiesStore = capabilityStore;
 		this._provider = provider;
 		this._textVirtualModel = textVirtualModel;
+		this._commandDetection = this._capabilitiesStore.get(TerminalCapability.CommandDetection);
+
 		// this.triggerCharacters = triggerCharacters;
 		// this.id = id;
+
+		this._registerCommandExecutionListener();
+
 	}
 
 	activate(terminal: Terminal): void {
 		console.log('activate');
 	}
 
+	private _registerCommandExecutionListener(): void {
+		// Listen to terminal command execution, update virtual document accordingly
+		if (this._commandDetection) {
+			this._commandDetection.onCommandFinished((e) => {
```

Comment:
This seems to never get triggered for me.
From debugging it seems like commandDetection is always undefined here 😅